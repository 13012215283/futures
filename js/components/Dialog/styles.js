import { StyleSheet } from 'react-native';

import { colors, sizes } from './../themes';
import { alpha } from '../../utils/color';
import { px2dp } from '../../utils/px2dp';

const headerTextColor = '#212121';
const textColorAnd = '#757575';

const style = StyleSheet.create({
  mask: {
    backgroundColor: alpha(colors.black, 0.4),
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dialogContainer: {
    backgroundColor: colors.white,
    marginLeft: px2dp(86),
    marginRight: px2dp(85),
    minWidth: px2dp(576),
    borderRadius: px2dp(6),
  },
  textContainer: {
    marginLeft: px2dp(48),
    marginTop: px2dp(48),
    marginRight: px2dp(53),
  },
  header: {
    fontSize: sizes.f4,
    color: headerTextColor,
    marginBottom: px2dp(48),
  },
  contentBox: {
    marginBottom: px2dp(32),
  },
  content: {
    fontSize: sizes.f2,
    color: textColorAnd,
    lineHeight: px2dp(48),
    marginBottom: px2dp(32),
  },
  activeContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginBottom: px2dp(16),
    height: px2dp(96),
    alignItems: 'center',
  },
  touchable: {
    borderColor: colors[1104],
    alignItems: 'center',
    justifyContent: 'center',
    width: px2dp(160),
    height: px2dp(72),
    marginRight: px2dp(32),
  },
  activeText: {
    flex: 1,
    fontSize: sizes.f3,
    textAlign: 'right',
    textAlignVertical: 'center',
    color: colors[1001],
  },
  text0: {
    color: colors[1102],
  },
});

export default style;
