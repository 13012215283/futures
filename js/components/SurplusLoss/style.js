import { StyleSheet } from 'react-native';
import { colors, sizes } from '../themes';
import { px2dp } from '../../utils/px2dp';

const style = StyleSheet.create({
  text: {
    color: colors[1102],
    fontSize: sizes.f2,
    marginBottom: px2dp(29.8),
  },
  surplusText: {
    marginTop: px2dp(31.2),
  },
});

export default style;
