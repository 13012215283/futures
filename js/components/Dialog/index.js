import React, { Component } from 'react';
import { Modal, Text, View, TouchableOpacity } from 'react-native';
import PropTypes, { oneOf, oneOfType } from 'prop-types';
import styles from './styles';

class Dialog extends Component {
  constructor(props) {
    super(props);
    this.renderButton = this.renderButton.bind(this);
    this.renderContent = this.renderContent.bind(this);
  }

  renderContent() {
    const { content } = this.props;

    if (typeof content === 'function') {
      return <View style={styles.contentBox}>{content()}</View>;
    }

    return content ? <Text style={styles.content}>{content}</Text> : null;
  }

  renderButton() {
    const { button } = this.props;

    // 最多支持传入两个按钮
    return button.slice(0, 2).map((v, k, array) => {
      let textStyle = [styles.activeText];
      if (array.length === 2)
        textStyle = [styles.activeText, styles[`text${k}`]];

      return (
        <TouchableOpacity
          key={v.name || ''}
          style={styles.touchable}
          onPress={() => v.callback()}
        >
          <Text style={[textStyle, v.style]}>{v.name}</Text>
        </TouchableOpacity>
      );
    });
  }

  render() {
    const { header, content, ...props } = this.props;
    return (
      <View>
        <Modal
          onRequestClose={this.props.onRequestClose}
          transparent
          {...props}
        >
          <View style={styles.mask}>
            <View style={styles.dialogContainer}>
              <View style={styles.textContainer}>
                <Text style={styles.header}>{header}</Text>
                {this.renderContent()}
              </View>
              <View style={styles.activeContainer}>{this.renderButton()}</View>
            </View>
          </View>
        </Modal>
      </View>
    );
  }
}

Dialog.propTypes = {
  header: PropTypes.string,
  content: oneOfType([PropTypes.string, PropTypes.func]),
  /* eslint-disable react/forbid-prop-types */
  button: PropTypes.array.isRequired,
  visible: PropTypes.bool,
  animationType: oneOf(['none', 'slide', 'fade']),
  onRequestClose: PropTypes.func,
};

Dialog.defaultProps = {
  header: '提示',
  visible: false,
  animationType: 'fade',
  onRequestClose() {},
  content: '',
};
export default Dialog;
