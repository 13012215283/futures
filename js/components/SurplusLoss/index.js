import React from 'react';
import { Text, View } from 'react-native';
import Stepper from '../Stepper';
import Dialog from '../Dialog';
import style from './style';

class SurplusLoss extends Dialog {
  static defaultProps = {
    step: 0.01,
    max: '0',
    min: '0',
  };

  constructor() {
    super();
    this.renderContent = this.renderContent.bind(this);
  }

  renderContent() {
    const { min, max, step } = this.props;

    return (
      <View>
        <Text style={style.text}>止损</Text>
        <Stepper
          value={min}
          step={step}
          inputViewStyle={{ flex: 1 }}
          inputItemStyle={{ width: '100%' }}
        />
        <Text style={[style.text, style.surplusText]}>止盈</Text>
        <Stepper
          value={max}
          step={step}
          inputViewStyle={{ flex: 1 }}
          inputItemStyle={{ width: '100%' }}
        />
      </View>
    );
  }

  render() {
    return <Dialog {...this.props} content={() => this.renderContent()} />;
  }
}

export default SurplusLoss;
