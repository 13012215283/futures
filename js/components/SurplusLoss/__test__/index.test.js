import React from 'react';
import { shallow } from 'enzyme';
import TestRender from 'react-test-renderer';
import Dialog from '../../Dialog';
import SurplusLoss from '../index';

describe('弹窗组件', () => {
  it('能够正确渲染', () => {
    const tree = TestRender.create(
      <SurplusLoss
        header="设置止盈止送"
        button={[
          {
            name: '取消',
            callback() {},
          },
          {
            name: '确认',
            callback() {},
          },
        ]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('组件包括一个Dialog', () => {
    const tree = shallow(
      <SurplusLoss
        header="设置止盈止送"
        button={[
          {
            name: '取消',
            callback() {},
          },
          {
            name: '确认',
            callback() {},
          },
        ]}
      />
    );

    expect(tree.find(Dialog).length).toBe(1);
  });
});
