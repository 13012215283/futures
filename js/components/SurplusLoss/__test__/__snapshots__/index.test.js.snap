// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`弹窗组件 能够正确渲染 1`] = `
<View>
  <Modal
    animationType="fade"
    button={
      Array [
        Object {
          "callback": [Function],
          "name": "取消",
        },
        Object {
          "callback": [Function],
          "name": "确认",
        },
      ]
    }
    hardwareAccelerated={false}
    max="0"
    min="0"
    onRequestClose={[Function]}
    step={0.01}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000066",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 6,
            "marginLeft": 86,
            "marginRight": 85,
            "minWidth": 576,
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": 48,
              "marginRight": 53,
              "marginTop": 48,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#212121",
                "fontSize": 40,
                "marginBottom": 48,
              }
            }
          >
            设置止盈止送
          </Text>
          <View
            style={
              Object {
                "marginBottom": 32,
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#757575",
                    "fontSize": 28,
                    "marginBottom": 29.8,
                  }
                }
              >
                止损
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#rgba(232,59,59,0.2)",
                        "borderColor": "#e83b3b",
                        "borderWidth": 1,
                        "height": 60,
                        "justifyContent": "center",
                        "width": 60,
                      },
                      null,
                    ]
                  }
                  testID={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#e83b3b",
                          "fontSize": 32,
                          "textAlign": "center",
                        },
                        null,
                      ]
                    }
                  >
                    -
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "marginLeft": 14,
                        "marginRight": 14,
                      },
                      Object {
                        "flex": 1,
                      },
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    inputStyle={
                      Object {
                        "width": "100%",
                      }
                    }
                    keyboardType="numeric"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="请输入"
                    style={
                      Array [
                        Object {
                          "borderColor": "#bdbdbd",
                          "borderWidth": 0.5,
                          "color": "#212121",
                          "fontSize": 28,
                          "height": 60,
                          "padding": 0,
                          "textAlign": "center",
                          "width": 250,
                        },
                        Object {
                          "width": "100%",
                        },
                      ]
                    }
                    thousandSeparator=","
                    type="money"
                    underlineColorAndroid="transparent"
                    value="0"
                  />
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#rgba(232,59,59,0.2)",
                        "borderColor": "#e83b3b",
                        "borderWidth": 1,
                        "height": 60,
                        "justifyContent": "center",
                        "width": 60,
                      },
                      null,
                    ]
                  }
                  testID={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#e83b3b",
                          "fontSize": 32,
                          "textAlign": "center",
                        },
                        null,
                      ]
                    }
                  >
                    +
                  </Text>
                </View>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#757575",
                      "fontSize": 28,
                      "marginBottom": 29.8,
                    },
                    Object {
                      "marginTop": 31.2,
                    },
                  ]
                }
              >
                止盈
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#rgba(232,59,59,0.2)",
                        "borderColor": "#e83b3b",
                        "borderWidth": 1,
                        "height": 60,
                        "justifyContent": "center",
                        "width": 60,
                      },
                      null,
                    ]
                  }
                  testID={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#e83b3b",
                          "fontSize": 32,
                          "textAlign": "center",
                        },
                        null,
                      ]
                    }
                  >
                    -
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "marginLeft": 14,
                        "marginRight": 14,
                      },
                      Object {
                        "flex": 1,
                      },
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    inputStyle={
                      Object {
                        "width": "100%",
                      }
                    }
                    keyboardType="numeric"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="请输入"
                    style={
                      Array [
                        Object {
                          "borderColor": "#bdbdbd",
                          "borderWidth": 0.5,
                          "color": "#212121",
                          "fontSize": 28,
                          "height": 60,
                          "padding": 0,
                          "textAlign": "center",
                          "width": 250,
                        },
                        Object {
                          "width": "100%",
                        },
                      ]
                    }
                    thousandSeparator=","
                    type="money"
                    underlineColorAndroid="transparent"
                    value="0"
                  />
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#rgba(232,59,59,0.2)",
                        "borderColor": "#e83b3b",
                        "borderWidth": 1,
                        "height": 60,
                        "justifyContent": "center",
                        "width": 60,
                      },
                      null,
                    ]
                  }
                  testID={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#e83b3b",
                          "fontSize": 32,
                          "textAlign": "center",
                        },
                        null,
                      ]
                    }
                  >
                    +
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 96,
              "justifyContent": "flex-end",
              "marginBottom": 16,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#e0e0e0",
                "height": 72,
                "justifyContent": "center",
                "marginRight": 32,
                "opacity": 1,
                "width": 160,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Array [
                    Object {
                      "color": "#e83b3b",
                      "flex": 1,
                      "fontSize": 32,
                      "textAlign": "right",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": "#757575",
                    },
                  ],
                  undefined,
                ]
              }
            >
              取消
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#e0e0e0",
                "height": 72,
                "justifyContent": "center",
                "marginRight": 32,
                "opacity": 1,
                "width": 160,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Array [
                    Object {
                      "color": "#e83b3b",
                      "flex": 1,
                      "fontSize": 32,
                      "textAlign": "right",
                      "textAlignVertical": "center",
                    },
                    undefined,
                  ],
                  undefined,
                ]
              }
            >
              确认
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
